src/reducer.js

```diff
@@ -250,7 +249,11 @@ export const authReducer = (state = { isLoaded: false, isEmpty: true }, action)
         }
       }
       const auth = action.auth.toJSON ? action.auth.toJSON() : action.auth
-      return { ...auth, isEmpty: false, isLoaded: true }
+      // Support keeping data
+      if (action.preserve && action.preserve.auth) {
+        return pick({ ...state, ...auth }, action.preserve.auth) // pick returns a new object
+      }
+      return { ...state, ...auth, isEmpty: false, isLoaded: true }
```

Comment:
before we were returning `{ ...auth, isEmpty: false, isLoaded: true }` and now we are returning `{ ...state, ...auth, isEmpty: false, isLoaded: true }`.  Any reason we added `...state`?

Reply:
Great question! The goal was to preserve anything in state that was there, as well as spread the new parameters onto it. I think that this doesn't really offer that much advantage, and could be unclear to some, so I am going to change it back.

I think it still makes sense to pass `{ ...state, ...auth }` to `pick` though since some want to be able to preserve `isLoaded` state and potentially other things that were already in state.